<center>
    <img src="https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/Logos/organization_logo/organization_logo.png" width="300" alt="cognitiveclass.ai logo"  />
</center>

# Working with a real world data-set using SQL and Python

Estaimted time needed: **30** minutes

## Objectives

After complting this lab you will be able to:

*   Understand the dataset for Chicago Public School level performance
*   Store the dataset in an Db2 database on IBM Cloud instance
*   Retrieve metadata about tables and columns and query data from mixed case columns
*   Solve example problems to practice your SQL skills including using built-in database functions


## Chicago Public Schools - Progress Report Cards (2011-2012)

The city of Chicago released a dataset showing all school level performance data used to create School Report Cards for the 2011-2012 school year. The dataset is available from the Chicago Data Portal: [https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t](https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

This dataset includes a large number of metrics. Start by familiarizing yourself with the types of metrics in the database: [https://data.cityofchicago.org/api/assets/AAD41A13-BE8A-4E67-B1F5-86E711E09D5F?download=true](https://data.cityofchicago.org/api/assets/AAD41A13-BE8A-4E67-B1F5-86E711E09D5F?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&download=true&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

**NOTE**:

Do not download the dataset directly from City of Chicago portal. Instead download a static copy which is a more database friendly version from this <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">link</a>.

Now review some of its contents.


### Store the dataset in a Table

In many cases the dataset to be analyzed is available as a .CSV (comma separated values) file, perhaps on the internet. To analyze the data using SQL, it first needs to be stored in the database.

While it is easier to read the dataset into a Pandas dataframe and then PERSIST it into the database as we saw in the previous lab, it results in mapping to default datatypes which may not be optimal for SQL querying. For example a long textual field may map to a CLOB instead of a VARCHAR.

Therefore, **it is highly recommended to manually load the table using the database console LOAD tool, as indicated in Week 2 Lab 1 Part II**. The only difference with that lab is that in Step 5 of the instructions you will need to click on create "(+) New Table" and specify the name of the table you want to create and then click "Next".

##### Now open the Db2 console, open the LOAD tool, Select / Drag the .CSV file for the CHICAGO PUBLIC SCHOOLS dataset and load the dataset into a new table called **SCHOOLS**.

<a href="https://cognitiveclass.ai/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01"><img src = "https://ibm.box.com/shared/static/uc4xjh1uxcc78ks1i18v668simioz4es.jpg"></a>


### Connect to the database

Let us now load the ipython-sql  extension and establish a connection with the database


In [2]:
!pip install sqlalchemy==1.3.9
!pip install ibm_db_sa

     |████████████████████████████████| 6.0 MB 29.5 MB/s eta 0:00:01
  Created wheel for sqlalchemy: filename=SQLAlchemy-1.3.9-cp38-cp38-linux_x86_64.whl size=1209508 sha256=c1b1e9c9751ce271d647babc1332fb756b1d02ae2a7fec7aea853985f323d64d
  Stored in directory: /tmp/wsuser/.cache/pip/wheels/cb/43/46/fa638f2422554332b7865d600275b24568bf60e76104a94bb4
Successfully built sqlalchemy
  Attempting uninstall: sqlalchemy
    Found existing installation: SQLAlchemy 1.4.22
    Uninstalling SQLAlchemy-1.4.22:
      Successfully uninstalled SQLAlchemy-1.4.22


In [3]:
pip install ipython-sql

     |████████████████████████████████| 42 kB 2.2 MB/s  eta 0:00:01
  Created wheel for prettytable: filename=prettytable-0.7.2-py3-none-any.whl size=13700 sha256=5d6417f604ae60536d8087c4303d0b36775d943ce9b29603b0562bae2b7932d1
  Stored in directory: /tmp/wsuser/.cache/pip/wheels/48/6d/77/9517cb933af254f51a446f1a5ec9c2be3e45f17384940bce68
Successfully built prettytable
Note: you may need to restart the kernel to use updated packages.


In [1]:
%load_ext sql

In [3]:
# Enter the connection string for your Db2 on Cloud database instance below
# %sql ibm_db_sa://my-username:my-password@my-hostname:my-port/my-db-name?security=SSL
%sql ibm_db_sa://hpy17939:QoLueplaozaEbSn1@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB?security=SSL

### Query the database system catalog to retrieve table metadata

##### You can verify that the table creation was successful by retrieving the list of all tables in your schema and checking whether the SCHOOLS table was created


In [6]:
# type in your query to retrieve list of all tables in the database for your db2 schema (username)
%sql select * from syscat.tables where tabschema = 'HPY17939'

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[('HPY17939', 'INTERNATIONAL_STUDENT_TEST_SCORES', 'HPY17939', 'U', 'T', 'N', None, None, None, None, datetime.datetime(2021, 11, 4, 18, 6, 23, 528967), datetime.datetime(2021, 11, 4, 18, 6, 23, 528967), datetime.datetime(2021, 11, 4, 18, 6, 23, 528967), datetime.datetime(2021, 11, 4, 18, 8, 8, 496783), 4, 11, 1928, 99, 1, 0, 1, -1, -1, -1, 0, 'HPY17939SPACE1', None, None, 0, 0, 0, 0, 0, 0, 0, 'N', 'YYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYY', 1, ' ', '0', -1, 'N', ' ', None, 'R', ' ', 'N', '                                ', None, 'N', ' ', 'F', None, 0, 'N', 999, 0, 0.0, 49, 0.0, None, 1208, 'SYSIBM', 'IDENTITY', 'SYSIBM', 'IDENTITY', ' ', 0, datetime.datetime(2021, 11, 4, 18, 6, 23, 528967), 0, ' ', None, None, 'N', 'HPY17939', ' ', ' ', ' ', datetime.date(2021, 11, 4), ' ', 'N', 'R', 'N', -1.0, None),
 ('HPY17939', 'CHICAGO_SOCIOECONOMIC_DATA', 'HPY17939', 'U', 'T', 'N', None, None, None, None, datetime.datetime(2021, 11, 4, 18, 22, 30, 231140), datetime.datetime(2021, 11, 4, 18, 22, 30, 231140), datetime.datetime(2021, 11, 4, 18, 22, 30, 491437), datetime.datetime(2021, 11, 4, 18, 22, 39, 536431), 10, 12, 1928, 78, 1, 0, 1, -1, -1, -1, 0, 'HPY17939SPACE1', None, None, 0, 0, 0, 0, 0, 0, 0, 'N', 'YYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYY', 1, ' ', '0', -1, 'N', ' ', None, 'R', ' ', 'N', '                                ', None, 'N', ' ', 'F', None, 0, 'N', 999, 0, 0.0, 111, 0.0, None, 1208, 'SYSIBM', 'IDENTITY', 'SYSIBM', 'IDENTITY', ' ', 0, datetime.datetime(2021, 11, 4, 18, 22, 30, 231140), 0, ' ', None, None, 'N', 'HPY17939', ' ', ' ', ' ', datetime.date(2021, 11, 4), ' ', 'N', 'R', 'N', -1.0, None),
 ('HPY17939', 'SCHOOLS', 'HPY17939', 'U', 'T', 'N', None, None, None, None, datetime.datetime(2021, 11, 8, 8, 56, 28, 907145), datetime.datetime(2021, 11, 8, 8, 56, 28, 907145), datetime.datetime(2021, 11, 8, 8, 56, 28, 907145), datetime.datetime(2021, 11, 8, 8, 57, 5, 263428), 78, 13, 1928, 566, 15, 0, 16, -1, -1, -1, 0, 'HPY17939SPACE1', None, None, 0, 0, 0, 0, 0, 0, 0, 'N', 'YYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYY', 1, ' ', '0', -1, 'N', ' ', None, 'R', ' ', 'N', '                                ', None, 'N', ' ', 'F', None, 0, 'N', 999, 0, 0.0, 843, 0.0, None, 1208, 'SYSIBM', 'IDENTITY', 'SYSIBM', 'IDENTITY', ' ', 0, datetime.datetime(2021, 11, 8, 8, 56, 28, 907145), 0, ' ', None, None, 'N', 'HPY17939', ' ', ' ', ' ', datetime.date(2021, 11, 8), ' ', 'N', 'R', 'N', -1.0, None),
 ('HPY17939', 'EMPLOYEES', 'HPY17939', 'U', 'T', 'N', None, None, None, None, datetime.datetime(2021, 10, 25, 11, 59, 19, 343659), datetime.datetime(2021, 10, 25, 11, 59, 19, 435465), datetime.datetime(2021, 10, 25, 11, 59, 19, 435469), datetime.datetime(2021, 10, 25, 12, 3, 39, 343602), 11, 5, 1928, 10, 1, 0, 2, -1, -1, -1, 0, 'HPY17939SPACE1', None, None, 0, 0, 0, 1, 1, 0, 0, 'N', 'YYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYY', 1, ' ', '0', -1, 'N', ' ', None, 'R', ' ', 'N', '                                ', None, 'N', ' ', 'F', None, 0, 'N', 999, 0, 0.0, 118, 0.0, None, 1208, 'SYSIBM', 'IDENTITY', 'SYSIBM', 'IDENTITY', ' ', 0, datetime.datetime(2021, 10, 25, 11, 59, 19, 343659), 0, ' ', None, None, 'N', 'HPY17939', ' ', ' ', ' ', datetime.date(2021, 11, 2), ' ', 'N', 'R', 'N', -1.0, None),
 ('HPY17939', 'JOB_HISTORY', 'HPY17939', 'U', 'T', 'N', None, None, None, None, datetime.datetime(2021, 10, 25, 11, 59, 19, 548099), datetime.datetime(2021, 10, 25, 11, 59, 19, 664298), datetime.datetime(2021, 10, 25, 11, 59, 19, 664302), datetime.datetime(2021, 10, 25, 12, 50, 38, 389219), 4, 6, 1928, 10, 1, 0, 2, -1, -1, -1, 0, 'HPY17939SPACE1', None, None, 0, 0, 0, 2, 1, 0, 0, 'N', 'YYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYY', 1, ' ', '0', -1, 'N', ' ', None, 'R', ' ', 'N', '                                ', None, 'N', ' ', 'F', None, 0, 'N', 999, 0, 0.0, 43, 0.0, None, 1208, 'SYSIBM', 'IDENTITY', 'SYSIBM', 'IDENTITY', ' ', 0, datetime.datetime(2021, 10, 25, 11, 59, 19, 548099), 0, ' ', None, None, 'N', 'HPY17939', ' ', ' ', ' ', datetime.date(2021, 10, 25), ' ', 'N', 'R', 'N', -1.0, None),
 ('HPY179

Double-click **here** for a hint

<!--
In Db2 the system catalog table called SYSCAT.TABLES contains the table metadata
-->


Double-click **here** for the solution.

<!-- Solution:

%sql select TABSCHEMA, TABNAME, CREATE_TIME from SYSCAT.TABLES where TABSCHEMA='YOUR-DB2-USERNAME'

or, you can retrieve list of all tables where the schema name is not one of the system created ones:

%sql select TABSCHEMA, TABNAME, CREATE_TIME from SYSCAT.TABLES \
      where TABSCHEMA not in ('SYSIBM', 'SYSCAT', 'SYSSTAT', 'SYSIBMADM', 'SYSTOOLS', 'SYSPUBLIC')
      
or, just query for a specifc table that you want to verify exists in the database
%sql select * from SYSCAT.TABLES where TABNAME = 'SCHOOLS'

-->


### Query the database system catalog to retrieve column metadata

##### The SCHOOLS table contains a large number of columns. How many columns does this table have?


In [23]:
%sql select count(*) from SYSCAT.COLUMNS where TABNAME = 'SCHOOLS'

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[(78,)]

Double-click **here** for a hint

<!--
In Db2 the system catalog table called SYSCAT.COLUMNS contains the column metadata
-->


Double-click **here** for the solution.

<!-- Solution:

%sql select count(*) from SYSCAT.COLUMNS where TABNAME = 'SCHOOLS'

-->


Now retrieve the the list of columns in SCHOOLS table and their column type (datatype) and length.


In [4]:
%sql select * from SYSCAT.COLUMNS where TABNAME = 'SCHOOLS'

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[('HPY17939', 'SCHOOLS', 'ENVIRONMENT_ICON', 20, 'SYSIBM  ', 'VARCHAR', 11, 0, 'OCTETS', 11, None, 'Y', 1208, 'SYSIBM', 'IDENTITY', ' ', ' ', 6, "'Very Weak'", "'NDA'", 11, None, 0, 20, 10, 0, None, None, None, None, None, None, None, None, ' ', 0, -1, 'N', 'N', ' ', None, 'O', None, -1.0, -1, -1, 6, None, None, 'N', 'N', 'N', -1, -1.0, None, None, 'N', None),
 ('HPY17939', 'SCHOOLS', 'ENVIRONMENT_SCORE', 21, 'SYSIBM  ', 'SMALLINT', 2, 0, None, None, None, 'Y', 0, None, None, ' ', ' ', 80, '99', '2', 3, None, 0, 20, 10, 53, None, None, None, None, None, None, None, None, ' ', 0, -1, 'N', 'N', ' ', None, 'O', None, -1.0, -1, -1, -1, None, None, 'N', 'N', 'N', -1, -1.0, None, None, 'N', None),
 ('HPY17939', 'SCHOOLS', 'INSTRUCTION_ICON', 22, 'SYSIBM  ', 'VARCHAR', 11, 0, 'OCTETS', 11, None, 'Y', 1208, 'SYSIBM', 'IDENTITY', ' ', ' ', 6, "'Very Weak'", "'NDA'", 11, None, 0, 20, 10, 0, None, None, None, None, None, None, None, None, ' ', 0, -1, 'N', 'N', ' ', None, 'O', None, -1.0, -1, -1, 6, None, None, 'N', 'N', 'N', -1, -1.0, None, None, 'N', None),
 ('HPY17939', 'SCHOOLS', 'INSTRUCTION_SCORE', 23, 'SYSIBM  ', 'SMALLINT', 2, 0, None, None, None, 'Y', 0, None, None, ' ', ' ', 83, '99', '11', 3, None, 0, 20, 10, 53, None, None, None, None, None, None, None, None, ' ', 0, -1, 'N', 'N', ' ', None, 'O', None, -1.0, -1, -1, -1, None, None, 'N', 'N', 'N', -1, -1.0, None, None, 'N', None),
 ('HPY17939', 'SCHOOLS', 'LEADERS_ICON', 24, 'SYSIBM  ', 'VARCHAR', 4, 0, 'OCTETS', 4, None, 'Y', 1208, 'SYSIBM', 'IDENTITY', ' ', ' ', 1, "'Weak'", "'Weak'", 9, None, 0, 20, 10, 0, None, None, None, None, None, None, None, None, ' ', 0, -1, 'N', 'N', ' ', None, 'O', None, -1.0, -1, -1, 4, None, None, 'N', 'N', 'N', -1, -1.0, None, None, 'N', None),
 ('HPY17939', 'SCHOOLS', 'LEADERS_SCORE', 25, 'SYSIBM  ', 'VARCHAR', 3, 0, 'OCTETS', 3, None, 'Y', 1208, 'SYSIBM', 'IDENTITY', ' ', ' ', 79, "'99'", "'14'", 7, None, 0, 20, 10, 0, None, None, None, None, None, None, None, None, ' ', 0, -1, 'N', 'N', ' ', None, 'O', None, -1.0, -1, -1, 2, None, None, 'N', 'N', 'N', -1, -1.0, None, None, 'N', None),
 ('HPY17939', 'SCHOOLS', 'TEACHERS_ICON', 26, 'SYSIBM  ', 'VARCHAR', 11, 0, 'OCTETS', 11, None, 'Y', 1208, 'SYSIBM', 'IDENTITY', ' ', ' ', 6, "'Very Weak'", "'NDA'", 10, None, 0, 20, 10, 0, None, None, None, None, None, None, None, None, ' ', 0, -1, 'N', 'N', ' ', None, 'O', None, -1.0, -1, -1, 5, None, None, 'N', 'N', 'N', -1, -1.0, None, None, 'N', None),
 ('HPY17939', 'SCHOOLS', 'TEACHERS_SCORE', 27, 'SYSIBM  ', 'VARCHAR', 3, 0, 'OCTETS', 3, None, 'Y', 1208, 'SYSIBM', 'IDENTITY', ' ', ' ', 77, "'99'", "'14'", 7, None, 0, 20, 10, 0, None, None, None, None, None, None, None, None, ' ', 0, -1, 'N', 'N', ' ', None, 'O', None, -1.0, -1, -1, 2, None, None, 'N', 'N', 'N', -1, -1.0, None, None, 'N', None),
 ('HPY17939', 'SCHOOLS', 'PARENT_ENGAGEMENT_ICON', 28, 'SYSIBM  ', 'VARCHAR', 7, 0, 'OCTETS', 7, None, 'Y', 1208, 'SYSIBM', 'IDENTITY', ' ', ' ', 4, "'Strong'", "'NDA'", 10, None, 0, 20, 10, 0, None, None, None, None, None, None, None, None, ' ', 0, -1, 'N', 'N', ' ', None, 'O', None, -1.0, -1, -1, 5, None, None, 'N', 'N', 'N', -1, -1.0, None, None, 'N', None),
 ('HPY17939', 'SCHOOLS', 'PARENT_ENGAGEMENT_SCORE', 29, 'SYSIBM  ', 'VARCHAR', 3, 0, 'OCTETS', 3, None, 'Y', 1208, 'SYSIBM', 'IDENTITY', ' ', ' ', 32, "'69'", "'39'", 7, None, 0, 20, 10, 0, None, None, None, None, None, None, None, None, ' ', 0, -1, 'N', 'N', ' ', None, 'O', None, -1.0, -1, -1, 2, None, None, 'N', 'N', 'N', -1, -1.0, None, None, 'N', None),
 ('HPY17939', 'SCHOOLS', 'PARENT_ENVIRONMENT_ICON', 30, 'SYSIBM  ', 'VARCHAR', 7, 0, 'OCTETS', 7, None, 'Y', 1208, 'SYSIBM', 'IDENTITY', ' ', ' ', 4, "'Strong'", "'NDA'", 10, None, 0, 20, 10, 0, None, None, None, None, None, None, None, None, ' ', 0, -1, 'N', 'N', ' ', None, 'O', None, -1.0, -1, -1, 5, None, None, 'N', 'N', 'N', -1, -1.0, None, None, 'N', None),
 ('HPY17939', 'SCHOOLS', 'PARENT_ENVIRONMENT_SCORE', 31, 'SYSIBM  ', 'VARCHAR', 3, 0, 'OCTETS',

In [31]:
# type in your query to retrieve all column names in the SCHOOLS table along with their datatypes and length
%sql select COLNAME, TYPENAME, LENGTH from SYSCAT.COLUMNS where TABNAME = 'SCHOOLS'

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[('ENVIRONMENT_ICON', 'VARCHAR', 11),
 ('ENVIRONMENT_SCORE', 'SMALLINT', 2),
 ('INSTRUCTION_ICON', 'VARCHAR', 11),
 ('INSTRUCTION_SCORE', 'SMALLINT', 2),
 ('LEADERS_ICON', 'VARCHAR', 4),
 ('LEADERS_SCORE', 'VARCHAR', 3),
 ('TEACHERS_ICON', 'VARCHAR', 11),
 ('TEACHERS_SCORE', 'VARCHAR', 3),
 ('PARENT_ENGAGEMENT_ICON', 'VARCHAR', 7),
 ('PARENT_ENGAGEMENT_SCORE', 'VARCHAR', 3),
 ('PARENT_ENVIRONMENT_ICON', 'VARCHAR', 7),
 ('PARENT_ENVIRONMENT_SCORE', 'VARCHAR', 3),
 ('AVERAGE_STUDENT_ATTENDANCE', 'VARCHAR', 6),
 ('RATE_OF_MISCONDUCTS__PER_100_STUDENTS_', 'DECIMAL', 5),
 ('AVERAGE_TEACHER_ATTENDANCE', 'VARCHAR', 6),
 ('INDIVIDUALIZED_EDUCATION_PROGRAM_COMPLIANCE_RATE', 'VARCHAR', 7),
 ('PK_2_LITERACY__', 'VARCHAR', 4),
 ('PK_2_MATH__', 'VARCHAR', 4),
 ('GR3_5_GRADE_LEVEL_MATH__', 'VARCHAR', 4),
 ('GR3_5_GRADE_LEVEL_READ__', 'VARCHAR', 4),
 ('GR3_5_KEEP_PACE_READ__', 'VARCHAR', 4),
 ('GR3_5_KEEP_PACE_MATH__', 'VARCHAR', 4),
 ('GR6_8_GRADE_LEVEL_MATH__', 'VARCHAR', 4),
 ('GR6_8_GRADE_LEVEL_READ__', 'VARCHAR', 4),
 ('GR6_8_KEEP_PACE_MATH_', 'VARCHAR', 4),
 ('GR6_8_KEEP_PACE_READ__', 'VARCHAR', 4),
 ('GR_8_EXPLORE_MATH__', 'VARCHAR', 4),
 ('GR_8_EXPLORE_READ__', 'VARCHAR', 4),
 ('ISAT_EXCEEDING_MATH__', 'DECIMAL', 4),
 ('ISAT_EXCEEDING_READING__', 'DECIMAL', 4),
 ('ISAT_VALUE_ADD_MATH', 'DECIMAL', 3),
 ('ISAT_VALUE_ADD_READ', 'DECIMAL', 3),
 ('ISAT_VALUE_ADD_COLOR_MATH', 'VARCHAR', 6),
 ('ISAT_VALUE_ADD_COLOR_READ', 'VARCHAR', 6),
 ('STUDENTS_TAKING__ALGEBRA__', 'VARCHAR', 4),
 ('STUDENTS_PASSING__ALGEBRA__', 'VARCHAR', 4),
 ('9th Grade EXPLORE (2009)', 'VARCHAR', 4),
 ('9th Grade EXPLORE (2010)', 'VARCHAR', 4),
 ('10th Grade PLAN (2009)', 'VARCHAR', 4),
 ('10th Grade PLAN (2010)', 'VARCHAR', 4),
 ('NET_CHANGE_EXPLORE_AND_PLAN', 'VARCHAR', 3),
 ('11th Grade Average ACT (2011)', 'VARCHAR', 4),
 ('SCHOOL_ID', 'INTEGER', 4),
 ('NAME_OF_SCHOOL', 'VARCHAR', 64),
 ('ELEMENTARY_MIDDLE_OR_HIGH_SCHOOL', 'VARCHAR', 2),
 ('STREET_ADDRESS', 'VARCHAR', 29),
 ('CITY', 'VARCHAR', 7),
 ('STATE', 'VARCHAR', 2),
 ('ZIP_CODE', 'INTEGER', 4),
 ('PHONE_NUMBER', 'VARCHAR', 14),
 ('LINK', 'VARCHAR', 78),
 ('NETWORK_MANAGER', 'VARCHAR', 40),
 ('COLLABORATIVE_NAME', 'VARCHAR', 34),
 ('ADEQUATE_YEARLY_PROGRESS_MADE_', 'VARCHAR', 3),
 ('TRACK_SCHEDULE', 'VARCHAR', 12),
 ('CPS_PERFORMANCE_POLICY_STATUS', 'VARCHAR', 16),
 ('CPS_PERFORMANCE_POLICY_LEVEL', 'VARCHAR', 15),
 ('HEALTHY_SCHOOL_CERTIFIED', 'VARCHAR', 3),
 ('SAFETY_ICON', 'VARCHAR', 11),
 ('SAFETY_SCORE', 'SMALLINT', 2),
 ('FAMILY_INVOLVEMENT_ICON', 'VARCHAR', 11),
 ('FAMILY_INVOLVEMENT_SCORE', 'VARCHAR', 3),
 ('NET_CHANGE_PLAN_AND_ACT', 'VARCHAR', 3),
 ('COLLEGE_ELIGIBILITY__', 'VARCHAR', 4),
 ('GRADUATION_RATE__', 'VARCHAR', 4),
 ('COLLEGE_ENROLLMENT_RATE__', 'VARCHAR', 4),
 ('COLLEGE_ENROLLMENT', 'SMALLINT', 2),
 ('GENERAL_SERVICES_ROUTE', 'SMALLINT', 2),
 ('FRESHMAN_ON_TRACK_RATE__', 'VARCHAR', 4),
 ('X_COORDINATE', 'DECIMAL', 13),
 ('Y_COORDINATE', 'DECIMAL', 13),
 ('LATITUDE', 'DECIMAL', 18),
 ('LONGITUDE', 'DECIMAL', 18),
 ('COMMUNITY_AREA_NUMBER', 'SMALLINT', 2),
 ('COMMUNITY_AREA_NAME', 'VARCHAR', 22),
 ('WARD', 'SMALLINT', 2),
 ('POLICE_DISTRICT', 'SMALLINT', 2),
 ('LOCATION', 'VARCHAR', 27)]

Double-click **here** for the solution.

<!-- Solution:

%sql select COLNAME, TYPENAME, LENGTH from SYSCAT.COLUMNS where TABNAME = 'SCHOOLS'

or

%sql select distinct(NAME), COLTYPE, LENGTH from SYSIBM.SYSCOLUMNS where TBNAME = 'SCHOOLS'

-->


### Questions

1.  Is the column name for the "SCHOOL ID" attribute in upper or mixed case?
2.  What is the name of "Community Area Name" column in your table? Does it have spaces?
3.  Are there any columns in whose names the spaces and paranthesis (round brackets) have been replaced by the underscore character "\_"?


In [7]:
%sql select distinct(COLNAME) from SYSCAT.COLUMNS where (TABNAME = 'SCHOOLS' and COLNAME like '%_%')

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[('10th Grade PLAN (2009)',),
 ('10th Grade PLAN (2010)',),
 ('11th Grade Average ACT (2011)',),
 ('9th Grade EXPLORE (2009)',),
 ('9th Grade EXPLORE (2010)',),
 ('ADEQUATE_YEARLY_PROGRESS_MADE_',),
 ('AVERAGE_STUDENT_ATTENDANCE',),
 ('AVERAGE_TEACHER_ATTENDANCE',),
 ('CITY',),
 ('COLLABORATIVE_NAME',),
 ('COLLEGE_ELIGIBILITY__',),
 ('COLLEGE_ENROLLMENT',),
 ('COLLEGE_ENROLLMENT_RATE__',),
 ('COMMUNITY_AREA_NAME',),
 ('COMMUNITY_AREA_NUMBER',),
 ('CPS_PERFORMANCE_POLICY_LEVEL',),
 ('CPS_PERFORMANCE_POLICY_STATUS',),
 ('ENVIRONMENT_ICON',),
 ('ENVIRONMENT_SCORE',),
 ('FAMILY_INVOLVEMENT_ICON',),
 ('FAMILY_INVOLVEMENT_SCORE',),
 ('FRESHMAN_ON_TRACK_RATE__',),
 ('GENERAL_SERVICES_ROUTE',),
 ('GR3_5_GRADE_LEVEL_MATH__',),
 ('GR3_5_GRADE_LEVEL_READ__',),
 ('GR3_5_KEEP_PACE_MATH__',),
 ('GR3_5_KEEP_PACE_READ__',),
 ('GR6_8_GRADE_LEVEL_MATH__',),
 ('GR6_8_GRADE_LEVEL_READ__',),
 ('GR6_8_KEEP_PACE_MATH_',),
 ('GR6_8_KEEP_PACE_READ__',),
 ('GRADUATION_RATE__',),
 ('GR_8_EXPLORE_MATH__',),
 ('GR_8_EXPLORE_READ__',),
 ('HEALTHY_SCHOOL_CERTIFIED',),
 ('INDIVIDUALIZED_EDUCATION_PROGRAM_COMPLIANCE_RATE',),
 ('INSTRUCTION_ICON',),
 ('INSTRUCTION_SCORE',),
 ('ISAT_EXCEEDING_MATH__',),
 ('ISAT_EXCEEDING_READING__',),
 ('ISAT_VALUE_ADD_COLOR_MATH',),
 ('ISAT_VALUE_ADD_COLOR_READ',),
 ('ISAT_VALUE_ADD_MATH',),
 ('ISAT_VALUE_ADD_READ',),
 ('LATITUDE',),
 ('LEADERS_ICON',),
 ('LEADERS_SCORE',),
 ('LINK',),
 ('LOCATION',),
 ('LONGITUDE',),
 ('NAME_OF_SCHOOL',),
 ('NETWORK_MANAGER',),
 ('NET_CHANGE_EXPLORE_AND_PLAN',),
 ('NET_CHANGE_PLAN_AND_ACT',),
 ('PARENT_ENGAGEMENT_ICON',),
 ('PARENT_ENGAGEMENT_SCORE',),
 ('PARENT_ENVIRONMENT_ICON',),
 ('PARENT_ENVIRONMENT_SCORE',),
 ('PHONE_NUMBER',),
 ('PK_2_LITERACY__',),
 ('PK_2_MATH__',),
 ('POLICE_DISTRICT',),
 ('RATE_OF_MISCONDUCTS__PER_100_STUDENTS_',),
 ('SAFETY_ICON',),
 ('SAFETY_SCORE',),
 ('SCHOOL_ID',),
 ('SCHOOL_TYPE',),
 ('STATE',),
 ('STREET_ADDRESS',),
 ('STUDENTS_PASSING__ALGEBRA__',),
 ('STUDENTS_TAKING__ALGEBRA__',),
 ('TEACHERS_ICON',),
 ('TEACHERS_SCORE',),
 ('TRACK_SCHEDULE',),
 ('WARD',),
 ('X_COORDINATE',),
 ('Y_COORDINATE',),
 ('ZIP_CODE',)]

In [7]:
%sql select distinct(COLNAME) from SYSCAT.COLUMNS where (TABNAME = 'SCHOOLS' and COLNAME like 'C%')

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[('CITY',),
 ('COLLABORATIVE_NAME',),
 ('COLLEGE_ELIGIBILITY__',),
 ('COLLEGE_ENROLLMENT',),
 ('COLLEGE_ENROLLMENT_RATE__',),
 ('COMMUNITY_AREA_NAME',),
 ('COMMUNITY_AREA_NUMBER',),
 ('CPS_PERFORMANCE_POLICY_LEVEL',),
 ('CPS_PERFORMANCE_POLICY_STATUS',)]

In [6]:
%sql select distinct(COLNAME) from SYSCAT.COLUMNS where (TABNAME = 'SCHOOLS' and COLNAME like 'S%')

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[('SAFETY_ICON',),
 ('SAFETY_SCORE',),
 ('SCHOOL_ID',),
 ('SCHOOL_TYPE',),
 ('STATE',),
 ('STREET_ADDRESS',),
 ('STUDENTS_PASSING__ALGEBRA__',),
 ('STUDENTS_TAKING__ALGEBRA__',)]

## Problems

### Problem 1

##### How many Elementary Schools are in the dataset?


In [5]:
%sql select * from schools LIMIT 5

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[(610038, 'Abraham Lincoln Elementary School', 'ES', '615 W Kemper Pl', 'Chicago', 'IL', 60614, '(773) 534-5720', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610038.pdf', 'Fullerton Elementary Network', 'NORTH-NORTHWEST SIDE COLLABORATIVE', 'No', 'Standard', 'Not on Probation', 'Level 1', 'Yes', 'Very Strong', 99, 'Very Strong', '99', 'Strong', 74, 'Strong', 66, 'Weak', '65', 'Strong', '70', 'Strong', '56', 'Average', '47', '96.00%', Decimal('2.0'), '96.40%', '95.80%', '80.1', '43.3', '89.6', '84.9', '60.7', '62.6', '81.9', '85.2', '52', '62.4', '66.3', '77.9', Decimal('69.7'), Decimal('64.4'), Decimal('0.2'), Decimal('0.9'), 'Yellow', 'Green', '67.1', '54.5', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 813, 33, 'NDA', Decimal('1171699.458'), Decimal('1915829.428'), Decimal('41.92449696'), Decimal('-87.64452163'), 7, 'LINCOLN PARK', 43, 18, '(41.92449696, -87.64452163)'),
 (610281, 'Adam Clayton Powell Paideia Community Academy Elementary School', 'ES', '7511 S South Shore Dr', 'Chicago', 'IL', 60649, '(773) 535-6650', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610281.pdf', 'Skyway Elementary Network', 'SOUTH SIDE COLLABORATIVE', 'No', 'Track_E', 'Not on Probation', 'Level 1', 'No', 'Average', 54, 'Strong', '66', 'Strong', 74, 'Very Strong', 84, 'Weak', '63', 'Strong', '76', 'Weak', '46', 'Average', '50', '95.60%', Decimal('15.7'), '95.30%', '100.00%', '62.4', '51.7', '21.9', '15.1', '29', '42.8', '38.5', '27.4', '44.8', '42.7', '14.1', '34.4', Decimal('16.8'), Decimal('16.5'), Decimal('0.7'), Decimal('1.4'), 'Green', 'Green', '17.2', '27.3', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 521, 46, 'NDA', Decimal('1196129.985'), Decimal('1856209.466'), Decimal('41.76032435'), Decimal('-87.55673627'), 43, 'SOUTH SHORE', 7, 4, '(41.76032435, -87.55673627)'),
 (610185, 'Adlai E Stevenson Elementary School', 'ES', '8010 S Kostner Ave', 'Chicago', 'IL', 60652, '(773) 535-2280', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610185.pdf', 'Midway Elementary Network', 'SOUTHWEST SIDE COLLABORATIVE', 'No', 'Standard', 'Not on Probation', 'Level 2', 'No', 'Strong', 61, 'NDA', 'NDA', 'Average', 50, 'Weak', 36, 'Weak', 'NDA', 'NDA', 'NDA', 'Average', '47', 'Weak', '41', '95.70%', Decimal('2.3'), '94.70%', '98.30%', '53.7', '26.6', '38.3', '34.7', '43.7', '57.3', '48.8', '39.2', '46.8', '44', '7.5', '21.9', Decimal('18.3'), Decimal('15.5'), Decimal('-0.9'), Decimal('-1.0'), 'Red', 'Red', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 1324, 44, 'NDA', Decimal('1148427.165'), Decimal('1851012.215'), Decimal('41.74711093'), Decimal('-87.73170248'), 70, 'ASHBURN', 13, 8, '(41.74711093, -87.73170248)'),
 (609993, 'Agustin Lara Elementary Academy', 'ES', '4619 S Wolcott Ave', 'Chicago', 'IL', 60609, '(773) 535-4389', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_609993.pdf', 'Pershing Elementary Network', 'SOUTHWEST SIDE COLLABORATIVE', 'No', 'Track_E', 'Not on Probation', 'Level 1', 'No', 'Average', 56, 'Average', '44', 'Average', 45, 'Weak', 37, 'Weak', '65', 'Average', '48', 'Average', '53', 'Strong', '58', '95.50%', Decimal('10.4'), '95.80%', '100.00%', '76.9', 'NDA', '26', '24.7', '61.8', '49.7', '39.2', '27.2', '69.7', '60.6', '9.1', '18.2', Decimal('11.1'), Decimal('9.6'), Decimal('0.9'), Decimal('2.4'), 'Green', 'Green', '42.9', '25', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 556, 42, 'NDA', Decimal('1164504.290'), Decimal('1873959.199'), Decimal('41.80975690'), Decimal('-87.67214460'), 61, 'NEW CITY', 20, 9, '(41.8097569, -87.6721446)'),
 (610513, 'Air Force Academy High School', 'HS', '3630 S Wells St', 'Chicago', 'IL', 60609, '(773) 535-1590', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610513.pdf', 'Southwest Side High School Network', 'SOUTHWEST SIDE COLLABORATIVE', 'NDA', 'Standard', 'Not on Probation'

In [8]:
%%sql
select count(*) from SCHOOLS where SCHOOL_TYPE = 'ES'

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[(462,)]

Double-click **here** for a hint

<!--
Which column specifies the school type e.g. 'ES', 'MS', 'HS'? ("Elementary School, Middle School, High School")
-->


Double-click **here** for another hint

<!--
Does the column name have mixed case, spaces or other special characters?
If so, ensure you use double quotes around the "Name of the Column"
-->


Double-click **here** for the solution.

<!-- Solution:

%sql select count(*) from SCHOOLS where school_type = 'ES'

Correct answer: 462

-->


### Problem 2

##### What is the highest Safety Score?


In [14]:
%sql select MAX(SAFETY_SCORE) as MAX_SAFETY_SCORE from schools

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[(99,)]

Double-click **here** for a hint

<!--
Use the MAX() function
-->


Double-click **here** for the solution.

<!-- Hint:

%sql select MAX(Safety_Score) AS MAX_SAFETY_SCORE from SCHOOLS

Correct answer: 99
-->


### Problem 3

##### Which schools have highest Safety Score?


In [15]:
%sql select NAME_OF_SCHOOL from SCHOOLS where SAFETY_SCORE = (select MAX(SAFETY_SCORE) from schools)

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[('Abraham Lincoln Elementary School',),
 ('Alexander Graham Bell Elementary School',),
 ('Annie Keller Elementary Gifted Magnet School',),
 ('Augustus H Burley Elementary School',),
 ('Edgar Allan Poe Elementary Classical School',),
 ('Edgebrook Elementary School',),
 ('Ellen Mitchell Elementary School',),
 ('James E McDade Elementary Classical School',),
 ('James G Blaine Elementary School',),
 ('LaSalle Elementary Language Academy',),
 ('Mary E Courtenay Elementary Language Arts Center',),
 ('Northside College Preparatory High School',),
 ('Northside Learning Center High School',),
 ('Norwood Park Elementary School',),
 ('Oriole Park Elementary School',),
 ('Sauganash Elementary School',),
 ('Stephen Decatur Classical Elementary School',),
 ('Talman Elementary School',),
 ('Wildwood Elementary School',)]

Double-click **here** for the solution.

<!-- Solution:
In the previous problem we found out that the highest Safety Score is 99, so we can use that as an input in the where clause:

%sql select Name_of_School, Safety_Score from SCHOOLS where Safety_Score = 99

or, a better way:

%sql select Name_of_School, Safety_Score from SCHOOLS where \
  Safety_Score= (select MAX(Safety_Score) from SCHOOLS)


Correct answer: several schools with with Safety Score of 99.
-->


### Problem 4

##### What are the top 10 schools with the highest "Average Student Attendance"?


In [16]:
%sql select NAME_OF_SCHOOL, AVERAGE_STUDENT_ATTENDANCE from SCHOOLS \
where AVERAGE_STUDENT_ATTENDANCE not like 'None' \
order by AVERAGE_STUDENT_ATTENDANCE desc \
limit 10

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[('John Charles Haines Elementary School', '98.40%'),
 ('James Ward Elementary School', '97.80%'),
 ('Edgar Allan Poe Elementary Classical School', '97.60%'),
 ('Rachel Carson Elementary School', '97.60%'),
 ('Orozco Fine Arts & Sciences Elementary School', '97.60%'),
 ('Annie Keller Elementary Gifted Magnet School', '97.50%'),
 ('Andrew Jackson Elementary Language Academy', '97.40%'),
 ('Lenart Elementary Regional Gifted Center', '97.40%'),
 ('Disney II Magnet School', '97.30%'),
 ('John H Vanderpoel Elementary Magnet School', '97.20%')]

In [18]:
%sql select NAME_OF_SCHOOL, AVERAGE_STUDENT_ATTENDANCE from SCHOOLS \
order by AVERAGE_STUDENT_ATTENDANCE desc nulls last \
limit 10

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[('John Charles Haines Elementary School', '98.40%'),
 ('James Ward Elementary School', '97.80%'),
 ('Edgar Allan Poe Elementary Classical School', '97.60%'),
 ('Orozco Fine Arts & Sciences Elementary School', '97.60%'),
 ('Rachel Carson Elementary School', '97.60%'),
 ('Annie Keller Elementary Gifted Magnet School', '97.50%'),
 ('Andrew Jackson Elementary Language Academy', '97.40%'),
 ('Lenart Elementary Regional Gifted Center', '97.40%'),
 ('Disney II Magnet School', '97.30%'),
 ('John H Vanderpoel Elementary Magnet School', '97.20%')]

Double-click **here** for the solution.

<!-- Solution:

%sql select Name_of_School, Average_Student_Attendance from SCHOOLS \
    order by Average_Student_Attendance desc nulls last limit 10 

-->


### Problem 5

##### Retrieve the list of 5 Schools with the lowest Average Student Attendance sorted in ascending order based on attendance


In [21]:
%sql select NAME_OF_SCHOOL, AVERAGE_STUDENT_ATTENDANCE from SCHOOLS \
order by AVERAGE_STUDENT_ATTENDANCE asc \
limit 5

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[('Richard T Crane Technical Preparatory High School', '57.90%'),
 ('Barbara Vick Early Childhood & Family Center', '60.90%'),
 ('Dyett High School', '62.50%'),
 ('Wendell Phillips Academy High School', '63.00%'),
 ('Orr Academy High School', '66.30%')]

Double-click **here** for the solution.

<!-- Solution:

%sql SELECT Name_of_School, Average_Student_Attendance  \
     from SCHOOLS \
     order by Average_Student_Attendance \
     fetch first 5 rows only

-->


In [20]:
%sql SELECT Name_of_School, Average_Student_Attendance  \
     from SCHOOLS \
     order by Average_Student_Attendance \
     fetch first 5 rows only

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[('Richard T Crane Technical Preparatory High School', '57.90%'),
 ('Barbara Vick Early Childhood & Family Center', '60.90%'),
 ('Dyett High School', '62.50%'),
 ('Wendell Phillips Academy High School', '63.00%'),
 ('Orr Academy High School', '66.30%')]

### Problem 6

##### Now remove the '%' sign from the above result set for Average Student Attendance column


In [33]:
%sql select NAME_OF_SCHOOL,  REPLACE(AVERAGE_STUDENT_ATTENDANCE, '%', ' ') as AVERAGE_STUDENT_ATTENDANCE  from SCHOOLS \
order by AVERAGE_STUDENT_ATTENDANCE asc \
limit 5

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[('Richard T Crane Technical Preparatory High School', '57.90 '),
 ('Barbara Vick Early Childhood & Family Center', '60.90 '),
 ('Dyett High School', '62.50 '),
 ('Wendell Phillips Academy High School', '63.00 '),
 ('Orr Academy High School', '66.30 ')]

Double-click **here** for a hint

<!--
Use the REPLACE() function to replace '%' with ''
See documentation for this function at:
https://www.ibm.com/support/knowledgecenter/en/SSEPGG_10.5.0/com.ibm.db2.luw.sql.ref.doc/doc/r0000843.html
-->


Double-click **here** for the solution.

<!-- Hint:

%sql SELECT Name_of_School, REPLACE(Average_Student_Attendance, '%', '') \
     from SCHOOLS \
     order by Average_Student_Attendance \
     fetch first 5 rows only

-->


### Problem 7

##### Which Schools have Average Student Attendance lower than 70%?


In [37]:
%%sql
select NAME_OF_SCHOOL, AVERAGE_STUDENT_ATTENDANCE
from SCHOOLS
where AVERAGE_STUDENT_ATTENDANCE < '70%'
order by AVERAGE_STUDENT_ATTENDANCE

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[('Richard T Crane Technical Preparatory High School', '57.90%'),
 ('Barbara Vick Early Childhood & Family Center', '60.90%'),
 ('Dyett High School', '62.50%'),
 ('Wendell Phillips Academy High School', '63.00%'),
 ('Orr Academy High School', '66.30%'),
 ('Manley Career Academy High School', '66.80%'),
 ('Chicago Vocational Career Academy High School', '68.80%'),
 ('Roberto Clemente Community Academy High School', '69.60%')]

Double-click **here** for a hint

<!--
The datatype of the "Average_Student_Attendance" column is varchar.
So you cannot use it as is in the where clause for a numeric comparison.
First use the CAST() function to cast it as a DECIMAL or DOUBLE
e.g. CAST("Column_Name" as DOUBLE)
or simply: DECIMAL("Column_Name")
-->


Double-click **here** for another hint

<!--
Don't forget the '%' age sign needs to be removed before casting
-->


Double-click **here** for the solution.

<!-- Solution:

%sql SELECT Name_of_School, Average_Student_Attendance  \
     from SCHOOLS \
     where CAST ( REPLACE(Average_Student_Attendance, '%', '') AS DOUBLE ) < 70 \
     order by Average_Student_Attendance
     
or,

%sql SELECT Name_of_School, Average_Student_Attendance  \
     from SCHOOLS \
     where DECIMAL ( REPLACE(Average_Student_Attendance, '%', '') ) < 70 \
     order by Average_Student_Attendance

-->


### Problem 8

##### Get the total College Enrollment for each Community Area


In [43]:
%%sql
select COLLEGE_ENROLLMENT, COMMUNITY_AREA_NAME
from SCHOOLS


 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[(813, 'LINCOLN PARK'),
 (521, 'SOUTH SHORE'),
 (1324, 'ASHBURN'),
 (556, 'NEW CITY'),
 (302, 'ARMOUR SQUARE'),
 (266, 'ALBANY PARK'),
 (4368, 'NORTH CENTER'),
 (620, 'WEST TOWN'),
 (232, 'NORTH CENTER'),
 (1023, 'ALBANY PARK'),
 (998, 'NORTH CENTER'),
 (476, 'NEW CITY'),
 (307, 'LAKE VIEW'),
 (398, 'WEST TOWN'),
 (525, 'ROSELAND'),
 (428, 'ROSELAND'),
 (801, 'HUMBOLDT PARK'),
 (274, 'BEVERLY'),
 (551, 'EAST GARFIELD PARK'),
 (171, 'NORTH LAWNDALE'),
 (264, 'CALUMET HEIGHTS'),
 (586, 'LOGAN SQUARE'),
 (539, 'ENGLEWOOD'),
 (114, 'WEST TOWN'),
 (730, 'WOODLAWN'),
 (556, 'NEAR WEST SIDE'),
 (339, 'WEST ENGLEWOOD'),
 (245, 'MOUNT GREENWOOD'),
 (669, 'WEST TOWN'),
 (436, 'GRAND BOULEVARD'),
 (550, 'KENWOOD'),
 (345, 'WEST ENGLEWOOD'),
 (905, 'SOUTH CHICAGO'),
 (539, 'NEW CITY'),
 (652, 'CHATHAM'),
 (784, 'DUNNING'),
 (459, 'CHATHAM'),
 (573, 'WEST ENGLEWOOD'),
 (490, 'ASHBURN'),
 (572, 'LAKE VIEW'),
 (316, 'AUSTIN'),
 (337, 'WOODLAWN'),
 (250, 'AUSTIN'),
 (398, 'AVALON PARK'),
 (204, 'MORGAN PARK'),
 (634, 'BELMONT CRAGIN'),
 (1683, 'LOWER WEST SIDE'),
 (302, 'ENGLEWOOD'),
 (301, 'ENGLEWOOD'),
 (429, 'WEST TOWN'),
 (350, 'WASHINGTON PARK'),
 (325, 'KENWOOD'),
 (137, 'CLEARING'),
 (340, 'HYDE PARK'),
 (549, 'HUMBOLDT PARK'),
 (691, 'BRIGHTON PARK'),
 (550, 'GRAND BOULEVARD'),
 (286, 'SOUTH DEERING'),
 (766, 'CHATHAM'),
 (792, 'BRIGHTON PARK'),
 (2366, 'IRVING PARK'),
 (644, 'AVONDALE'),
 (343, 'ENGLEWOOD'),
 (537, 'ASHBURN'),
 (329, 'GRAND BOULEVARD'),
 (915, 'NEW CITY'),
 (1415, 'BELMONT CRAGIN'),
 (315, 'NORTH LAWNDALE'),
 (1032, 'ASHBURN'),
 (446, 'SOUTH LAWNDALE'),
 (246, 'WASHINGTON HEIGHTS'),
 (403, 'HYDE PARK'),
 (600, 'BRIDGEPORT'),
 (291, 'AVALON PARK'),
 (1890, 'BELMONT CRAGIN'),
 (622, 'LOGAN SQUARE'),
 (407, 'WEST ENGLEWOOD'),
 (310, 'GREATER GRAND CROSSING'),
 (551, 'GREATER GRAND CROSSING'),
 (430, 'WEST GARFIELD PARK'),
 (331, 'WEST ENGLEWOOD'),
 (601, 'DUNNING'),
 (502, 'DUNNING'),
 (579, 'MOUNT GREENWOOD'),
 (483, 'DOUGLAS'),
 (833, 'AVALON PARK'),
 (650, 'EDISON PARK'),
 (688, 'ROSELAND'),
 (309, 'WEST TOWN'),
 (602, 'AUBURN GRESHAM'),
 (531, 'CHICAGO LAWN'),
 (409, 'NORTH LAWNDALE'),
 (1038, 'BRIGHTON PARK'),
 (233, 'ROSELAND'),
 (274, 'GRAND BOULEVARD'),
 (298, 'NORTH LAWNDALE'),
 (894, 'SOUTH LAWNDALE'),
 (752, 'WEST RIDGE'),
 (160, 'NORWOOD PARK'),
 (324, 'GRAND BOULEVARD'),
 (631, 'SOUTH LAWNDALE'),
 (828, 'HUMBOLDT PARK'),
 (365, 'ENGLEWOOD'),
 (370, 'WEST GARFIELD PARK'),
 (1456, 'SOUTH LAWNDALE'),
 (200, 'NORTH CENTER'),
 (357, 'NEW CITY'),
 (1139, 'WEST RIDGE'),
 (363, 'IRVING PARK'),
 (823, 'CHICAGO LAWN'),
 (632, 'EAST SIDE'),
 (1001, 'BELMONT CRAGIN'),
 (915, 'KENWOOD'),
 (381, 'WOODLAWN'),
 (310, 'ROSELAND'),
 (543, 'ASHBURN'),
 (556, 'NORTH LAWNDALE'),
 (318, 'WASHINGTON PARK'),
 (192, 'PULLMAN'),
 (468, 'FOREST GLEN'),
 (431, 'NORWOOD PARK'),
 (244, 'WASHINGTON PARK'),
 (783, 'SOUTH SHORE'),
 (1365, 'WASHINGTON PARK'),
 (414, 'WEST GARFIELD PARK'),
 (554, 'SOUTH CHICAGO'),
 (321, 'MCKINLEY PARK'),
 (314, 'NEAR NORTH SIDE'),
 (377, 'AUSTIN'),
 (830, 'WEST LAWN'),
 (461, 'NEW CITY'),
 (138, 'WEST PULLMAN'),
 (1716, 'PORTAGE PARK'),
 (419, 'WEST ENGLEWOOD'),
 (182, 'GREATER GRAND CROSSING'),
 (1133, 'SOUTH LAWNDALE'),
 (762, 'BEVERLY'),
 (289, 'WEST TOWN'),
 (451, 'LINCOLN SQUARE'),
 (1111, 'AUSTIN'),
 (473, 'NEW CITY'),
 (344, 'WEST TOWN'),
 (458, 'GREATER GRAND CROSSING'),
 (988, 'SOUTH LAWNDALE'),
 (471, 'WOODLAWN'),
 (238, 'SOUTH SHORE'),
 (1021, 'GAGE PARK'),
 (581, 'GAGE PARK'),
 (672, 'PORTAGE PARK'),
 (342, 'MORGAN PARK'),
 (375, 'ROGERS PARK'),
 (786, 'OHARE'),
 (389, 'MCKINLEY PARK'),
 (604, 'CHICAGO LAWN'),
 (860, 'AVONDALE'),
 (100, 'NEAR WEST SIDE'),
 (1560, 'WEST ELSDON'),
 (344, 'WASHINGTON HEIGHTS'),
 (112, 'KENWOOD'),
 (1382, 'GAGE PARK'),
 (478, 'WASHINGTON HEIGHTS'),
 (418, 'AUBURN GRESHAM'),
 (314, 'SOUTH LAWNDALE'),
 (891, 'CHICAGO LAWN'),
 (224, 'FULLER PARK'),
 (301, 'AUSTIN'),
 (808, 'ENGLEWOOD'),
 (504, 'ROSELAND'),
 (609, 'ROSELAND'),
 (363, 'NEAR NORTH SIDE'),
 (867, 'BRIGHTON

In [45]:
%%sql
select SUM(COLLEGE_ENROLLMENT) as TOTAL_COLLEGE_ENROLLMENT, COMMUNITY_AREA_NAME
from SCHOOLS
group by COMMUNITY_AREA_NAME

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[(6864, 'ALBANY PARK'),
 (4823, 'ARCHER HEIGHTS'),
 (1458, 'ARMOUR SQUARE'),
 (6483, 'ASHBURN'),
 (4175, 'AUBURN GRESHAM'),
 (10933, 'AUSTIN'),
 (1522, 'AVALON PARK'),
 (3640, 'AVONDALE'),
 (14386, 'BELMONT CRAGIN'),
 (1636, 'BEVERLY'),
 (3167, 'BRIDGEPORT'),
 (9647, 'BRIGHTON PARK'),
 (549, 'BURNSIDE'),
 (1568, 'CALUMET HEIGHTS'),
 (5042, 'CHATHAM'),
 (7086, 'CHICAGO LAWN'),
 (2085, 'CLEARING'),
 (4670, 'DOUGLAS'),
 (4568, 'DUNNING'),
 (5337, 'EAST GARFIELD PARK'),
 (5305, 'EAST SIDE'),
 (4600, 'EDGEWATER'),
 (910, 'EDISON PARK'),
 (6832, 'ENGLEWOOD'),
 (1431, 'FOREST GLEN'),
 (531, 'FULLER PARK'),
 (9915, 'GAGE PARK'),
 (4552, 'GARFIELD RIDGE'),
 (2809, 'GRAND BOULEVARD'),
 (4051, 'GREATER GRAND CROSSING'),
 (963, 'HEGEWISCH'),
 (3975, 'HERMOSA'),
 (8620, 'HUMBOLDT PARK'),
 (1930, 'HYDE PARK'),
 (7764, 'IRVING PARK'),
 (1755, 'JEFFERSON PARK'),
 (4287, 'KENWOOD'),
 (7055, 'LAKE VIEW'),
 (5615, 'LINCOLN PARK'),
 (4132, 'LINCOLN SQUARE'),
 (7351, 'LOGAN SQUARE'),
 (871, 'LOOP'),
 (7257, 'LOWER WEST SIDE'),
 (1552, 'MCKINLEY PARK'),
 (1317, 'MONTCLARE'),
 (3271, 'MORGAN PARK'),
 (2091, 'MOUNT GREENWOOD'),
 (3362, 'NEAR NORTH SIDE'),
 (1378, 'NEAR SOUTH SIDE'),
 (7975, 'NEAR WEST SIDE'),
 (7922, 'NEW CITY'),
 (7541, 'NORTH CENTER'),
 (5146, 'NORTH LAWNDALE'),
 (4210, 'NORTH PARK'),
 (6469, 'NORWOOD PARK'),
 (140, 'OAKLAND'),
 (786, 'OHARE'),
 (6954, 'PORTAGE PARK'),
 (1620, 'PULLMAN'),
 (1547, 'RIVERDALE'),
 (4068, 'ROGERS PARK'),
 (7020, 'ROSELAND'),
 (4043, 'SOUTH CHICAGO'),
 (1859, 'SOUTH DEERING'),
 (14793, 'SOUTH LAWNDALE'),
 (4543, 'SOUTH SHORE'),
 (4388, 'UPTOWN'),
 (4006, 'WASHINGTON HEIGHTS'),
 (2648, 'WASHINGTON PARK'),
 (3700, 'WEST ELSDON'),
 (5946, 'WEST ENGLEWOOD'),
 (2622, 'WEST GARFIELD PARK'),
 (4207, 'WEST LAWN'),
 (3240, 'WEST PULLMAN'),
 (8197, 'WEST RIDGE'),
 (9429, 'WEST TOWN'),
 (4206, 'WOODLAWN')]

Double-click **here** for a hint

<!--
Verify the exact name of the Enrollment column in the database
Use the SUM() function to add up the Enrollments for each Community Area
-->


Double-click **here** for another hint

<!--
Don't forget to group by the Community Area
-->


Double-click **here** for the solution.

<!-- Solution:

%sql select Community_Area_Name, sum(College_Enrollment) AS TOTAL_ENROLLMENT \
   from SCHOOLS \
   group by Community_Area_Name 

-->


### Problem 9

##### Get the 5 Community Areas with the least total College Enrollment  sorted in ascending order


In [49]:
%%sql
select SUM(COLLEGE_ENROLLMENT) as TOTAL_COLLEGE_ENROLLMENT, COMMUNITY_AREA_NAME
from SCHOOLS
group by COMMUNITY_AREA_NAME
order by SUM(COLLEGE_ENROLLMENT)
limit 5

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[(140, 'OAKLAND'),
 (531, 'FULLER PARK'),
 (549, 'BURNSIDE'),
 (786, 'OHARE'),
 (871, 'LOOP')]

Double-click **here** for a hint

<!--
Order the previous query and limit the number of rows you fetch
-->


Double-click **here** for the solution.

<!-- Solution:

%sql select Community_Area_Name, sum(College_Enrollment) AS TOTAL_ENROLLMENT \
   from SCHOOLS \
   group by Community_Area_Name \
   order by TOTAL_ENROLLMENT asc \
   fetch first 5 rows only

-->


### Problem 10

##### List 5 schools with lowest safety score.


In [50]:
%%sql
select NAME_OF_SCHOOL, SAFETY_SCORE
from SCHOOLS
order by SAFETY_SCORE asc
limit 5

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[('Edmond Burke Elementary School', 1),
 ("Luke O'Toole Elementary School", 5),
 ('George W Tilton Elementary School', 6),
 ('Foster Park Elementary School', 11),
 ('Emil G Hirsch Metropolitan High School', 13)]

Double-click **here** for the solution.

<!-- Solution:

%sql SELECT name_of_school, safety_score
FROM schools
ORDER BY safety_score
LIMIT 5
-->


### Problem 11

##### Get the hardship index for the community area which has College Enrollment of 4368


In [56]:
%sql select distinct(COLNAME) from SYSCAT.COLUMNS where TABNAME = 'SCHOOLS'

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[('10th Grade PLAN (2009)',),
 ('10th Grade PLAN (2010)',),
 ('11th Grade Average ACT (2011)',),
 ('9th Grade EXPLORE (2009)',),
 ('9th Grade EXPLORE (2010)',),
 ('ADEQUATE_YEARLY_PROGRESS_MADE_',),
 ('AVERAGE_STUDENT_ATTENDANCE',),
 ('AVERAGE_TEACHER_ATTENDANCE',),
 ('CITY',),
 ('COLLABORATIVE_NAME',),
 ('COLLEGE_ELIGIBILITY__',),
 ('COLLEGE_ENROLLMENT',),
 ('COLLEGE_ENROLLMENT_RATE__',),
 ('COMMUNITY_AREA_NAME',),
 ('COMMUNITY_AREA_NUMBER',),
 ('CPS_PERFORMANCE_POLICY_LEVEL',),
 ('CPS_PERFORMANCE_POLICY_STATUS',),
 ('ENVIRONMENT_ICON',),
 ('ENVIRONMENT_SCORE',),
 ('FAMILY_INVOLVEMENT_ICON',),
 ('FAMILY_INVOLVEMENT_SCORE',),
 ('FRESHMAN_ON_TRACK_RATE__',),
 ('GENERAL_SERVICES_ROUTE',),
 ('GR3_5_GRADE_LEVEL_MATH__',),
 ('GR3_5_GRADE_LEVEL_READ__',),
 ('GR3_5_KEEP_PACE_MATH__',),
 ('GR3_5_KEEP_PACE_READ__',),
 ('GR6_8_GRADE_LEVEL_MATH__',),
 ('GR6_8_GRADE_LEVEL_READ__',),
 ('GR6_8_KEEP_PACE_MATH_',),
 ('GR6_8_KEEP_PACE_READ__',),
 ('GRADUATION_RATE__',),
 ('GR_8_EXPLORE_MATH__',),
 ('GR_8_EXPLORE_READ__',),
 ('HEALTHY_SCHOOL_CERTIFIED',),
 ('INDIVIDUALIZED_EDUCATION_PROGRAM_COMPLIANCE_RATE',),
 ('INSTRUCTION_ICON',),
 ('INSTRUCTION_SCORE',),
 ('ISAT_EXCEEDING_MATH__',),
 ('ISAT_EXCEEDING_READING__',),
 ('ISAT_VALUE_ADD_COLOR_MATH',),
 ('ISAT_VALUE_ADD_COLOR_READ',),
 ('ISAT_VALUE_ADD_MATH',),
 ('ISAT_VALUE_ADD_READ',),
 ('LATITUDE',),
 ('LEADERS_ICON',),
 ('LEADERS_SCORE',),
 ('LINK',),
 ('LOCATION',),
 ('LONGITUDE',),
 ('NAME_OF_SCHOOL',),
 ('NETWORK_MANAGER',),
 ('NET_CHANGE_EXPLORE_AND_PLAN',),
 ('NET_CHANGE_PLAN_AND_ACT',),
 ('PARENT_ENGAGEMENT_ICON',),
 ('PARENT_ENGAGEMENT_SCORE',),
 ('PARENT_ENVIRONMENT_ICON',),
 ('PARENT_ENVIRONMENT_SCORE',),
 ('PHONE_NUMBER',),
 ('PK_2_LITERACY__',),
 ('PK_2_MATH__',),
 ('POLICE_DISTRICT',),
 ('RATE_OF_MISCONDUCTS__PER_100_STUDENTS_',),
 ('SAFETY_ICON',),
 ('SAFETY_SCORE',),
 ('SCHOOL_ID',),
 ('SCHOOL_TYPE',),
 ('STATE',),
 ('STREET_ADDRESS',),
 ('STUDENTS_PASSING__ALGEBRA__',),
 ('STUDENTS_TAKING__ALGEBRA__',),
 ('TEACHERS_ICON',),
 ('TEACHERS_SCORE',),
 ('TRACK_SCHEDULE',),
 ('WARD',),
 ('X_COORDINATE',),
 ('Y_COORDINATE',),
 ('ZIP_CODE',)]

In [67]:
%sql select distinct(COLNAME) from SYSCAT.COLUMNS where TABNAME = 'CHICAGO_SOCIOECONOMIC_DATA'

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[('CA',),
 ('COMMUNITY_AREA_NAME',),
 ('HARDSHIP_INDEX',),
 ('PERCENT_AGED_16_UNEMPLOYED',),
 ('PERCENT_AGED_25_WITHOUT_HIGH_SCHOOL_DIPLOMA',),
 ('PERCENT_AGED_UNDER_18_OR_OVER_64',),
 ('PERCENT_HOUSEHOLDS_BELOW_POVERTY',),
 ('PERCENT_OF_HOUSING_CROWDED',),
 ('PER_CAPITA_INCOME_',),
 ('index',)]

In [68]:
%%sql
select COMMUNITY_AREA_NAME, COLLEGE_ENROLLMENT
from SCHOOLS
where COLLEGE_ENROLLMENT = 4368


 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[('NORTH CENTER', 4368)]

In [96]:
%%sql
select CSD.HARDSHIP_INDEX
from SCHOOLS S, CHICAGO_SOCIOECONOMIC_DATA CSD
where  S.COMMUNITY_AREA_NUMBER = CSD.CA and COLLEGE_ENROLLMENT = 4368

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[(6.0,)]

Double-click **here** for the solution.

<!-- Solution:
NOTE: For this solution to work the CHICAGO_SOCIOECONOMIC_DATA table 
      as created in the last lab of Week 3 should already exist

%%sql 
select hardship_index 
   from chicago_socioeconomic_data CD, schools CPS 
   where CD.ca = CPS.community_area_number 
      and college_enrollment = 4368

-->


In [74]:
%%sql 
select hardship_index 
   from chicago_socioeconomic_data CD, schools CPS 
   where CD.ca = CPS.community_area_number 
      and college_enrollment = 4368

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[(6.0,)]

### Problem 12

##### Get the hardship index for the community area which has the school with the  highest enrollment.


In [97]:
%%sql
select COMMUNITY_AREA_NAME, COLLEGE_ENROLLMENT, NAME_OF_SCHOOL
from SCHOOLS
order by COLLEGE_ENROLLMENT desc limit 1

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[('NORTH CENTER', 4368, 'Albert G Lane Technical High School')]

In [102]:
%%sql
select ca, community_area_name, hardship_index
from chicago_socioeconomic_data
where ca in  (select COMMUNITY_AREA_NUMBER
from SCHOOLS
order by COLLEGE_ENROLLMENT desc limit 1)

 * ibm_db_sa://hpy17939:***@2d46b6b4-cbf6-40eb-bbce-6251e6ba0300.bs2io90l08kqb1od8lcg.databases.appdomain.cloud:32328/BLUDB
Done.


[(5.0, 'North Center', 6.0)]

Double-click **here** for the solution.

<!-- Solution:
NOTE: For this solution to work the CHICAGO_SOCIOECONOMIC_DATA table 
      as created in the last lab of Week 3 should already exist

%sql select ca, community_area_name, hardship_index from chicago_socioeconomic_data \
   where ca in \
   ( select community_area_number from schools order by college_enrollment desc limit 1 )

-->


## Summary

##### In this lab you learned how to work with a real word dataset using SQL and Python. You learned how to query columns with spaces or special characters in their names and with mixed case names. You also used built in database functions and practiced how to sort, limit, and order result sets, as well as used sub-queries and worked with multiple tables.


## Author

<a href="https://www.linkedin.com/in/ravahuja/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Rav Ahuja</a>

## Change Log

| Date (YYYY-MM-DD) | Version | Changed By        | Change Description                        |
| ----------------- | ------- | ----------------- | ----------------------------------------- |
| 2021-07-09        | 2.4     | Malika            | Updated connection string                 |
| 2021-05-19        | 2.3     | Lakshmi Holla     | Updated question                          |
| 2021-04-20        | 2.2     | Malika            | Added the libraries                       |
| 2020-11-27        | 2.1     | Sannareddy Ramesh | Modified data sets and added new problems |
| 2020-08-28        | 2.0     | Lavanya           | Moved lab to course repo in GitLab        |

<hr>

## <h3 align="center"> © IBM Corporation 2020. All rights reserved. <h3/>
